readers-frontend/src/components/FeaturedArticles.astro

```diff
@@ -0,0 +1,380 @@
+---
+import type { categories, featuredHome } from "@prisma/client";
+import prisma from "../prisma";
+import Article from './Article.astro';
+import { getArticles, articlesWithArticleDrafts, getArticleImage } from "./utils/articles";
+import { getMainArticleCategory } from "./utils/categories";
+
+interface Props {
+  featuredArticlesIds: number[] | undefined;
+}
+
+const { featuredArticlesIds} = Astro.props;
+
+// Get featured articles
+const featuredArticles: articlesWithArticleDrafts[] = await getArticles(featuredArticlesIds)
+const hasPortraitArticles = featuredArticles.filter(a => a.articles_isThumbnailPortrait).length > 0;
+// Sort articles so that portrait articles are first
+if (hasPortraitArticles) {
+  featuredArticles.sort((a, b) => {
+    if (a.articles_isThumbnailPortrait && !b.articles_isThumbnailPortrait) {
+      return -1;
+    } else {
+      return 1;
+    }
+  });
+}
+
+// Get the main categories for each article
+const hexEncode = function(x: string){
+    var hex, i;
+
+    var result = "";
+    for (i=0; i<x.length; i++) {
+        hex = x.charCodeAt(i).toString(16);
+        result += ("000"+hex).slice(-4);
+    }
+
+    return result
+}
+const mainCategories: (categories | null)[] = await Promise.all(featuredArticles.map(article => getMainArticleCategory(article.articles_categories.split(',').map(Number))))
+---
+{!hasPortraitArticles &&
+  <!-- 4 or more Articles -->
```

Comment:
Can we have these as Astro comments so they don't all show up in the generated html?

Reply:
Ahh I didn't know about that. I fixed it now :)